docs/gh-action.rst

```diff
@@ -0,0 +1,136 @@
+* Create a file with ``.yml`` extension.
+* Enter the following code in the YAML file:
+
+.. code:: yaml
+
+    name: Setup Bowtie
+    run-name: ${{ github.actor }} is learning to use Bowtie
+
+    on: [push]
+
+    jobs:
+    sets-up-bowtie:
+        runs-on: ubuntu-latest
+
+        steps:
+        - name: Install Bowtie
+            uses: bowtie-json-schema/bowtie@v2023.05.12
+
+
+*The code snippet will install bowtie every time you push onto the repository.*
+
+2. Verification Steps
+^^^^^^^^^^^^^^^^^^^^^
+
+* Append the following code in the already created YAML file:
+
+.. code:: yaml
+
+        - name: Run Bowtie
+          run: bowtie --version
+
+
+* The YAML file will look something like this:
+
+.. code:: yaml
+
+    name: Setup Bowtie
+    run-name: ${{ github.actor }} is learning to use Bowtie
+
+    on: [push]
+
+    jobs:
+        sets-up-bowtie:
+            runs-on: ubuntu-latest
+
+            steps:
+            - name: Install Bowtie
+              uses: bowtie-json-schema/bowtie@v2023.05.12
+
+            - name: Run Bowtie
+              run: bowtie --version
+
+* Push the changes to your repository.
+* Go to the ``Actions`` tab of your repository and wait for the action to complete.
+* If the run was successful, you will see a green circle with a tick.
+* Click on the Workflow and then click on the ``sets-up-bowtie`` Job.
+* Open the drop down of the ``Run Bowtie`` step.
+* If it shows something like this, bowtie is running successfully:
+
+.. code:: sh
+
+    bowtie, version 2023.6.4
+
+.. admonition:: Note
+
+    IF THE RUN WAS NOT SUCCESSFUL, YOU WILL SEE A RED CIRCLE WITH A CROSS. RECHECK THE CODE YOU HAVE WRITTEN, AND CORRECT IN CASE OF ANY DISCREPANCY.
+
+*This will help us to test if bowtie is working in the GitHub action.*
+
+
+Validate using bowtie
+---------------------
+
+Now that you have successfully added bowtie to your workflow, let's work on using it to validate your JSON Specifications.
+
+1. Add the ``validate`` Command
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Add the following code snippet to your YAML file:
+
+.. code:: yaml
+
+        - name: Validate Schema
+          run: bowtie validate -i lua-jsonschema schema.json instance.json
+
+Let's break down this command:
+
+* ``bowtie validate``: It tells which command to run.
+* ``-i``: It is a required flag for ``validate`` command and specifies which implementation to be used, which in this case is ``lua-jsonschema``.
+* ``schema.json``: It is the name of the file containing the defined schema against which instances will be validated.
+* ``instance.json``: It is the name of the file containing the instances to be validated. This can also be the path to a directory that contains all the instances.
+
+
+*You will see that the implementation is skipped and thus does not validate the instances. This is because the lua implementation does not support the default 2020-12 draft.*
+
+
+2. Change the Dialect
```

Comment:
This sort of information I think belongs in the CLI documentation, it's not specific to using Bowtie in GitHub actions, so perhaps have a look there and see if it already contains this info -- if it doesn't, we should add it there, but if it does, you can probably just link to that page instead of repeating it here.

Reply:
I wrote it here specifically because I did not find it in the CLI docs. I will add it to that page instead of here, I agree with it being more suited there.